pom.xml

```diff
@@ -341,21 +341,41 @@ under the License.
                         <goals>
                             <goal>java</goal>
                         </goals>
+
+                        <configuration>
+                            <mainClass>org.apache.flink.tools.ci.licensecheck.LicenseChecker</mainClass>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>check-shade</id>
+                        <!-- manually called -->
+                        <phase>none</phase>
+                        <goals>
+                            <goal>java</goal>
+                        </goals>
+
+                        <configuration>
+                            <mainClass>org.apache.flink.tools.ci.optional.ShadeOptionalChecker</mainClass>
+                        </configuration>
                     </execution>
                 </executions>
                 <configuration>
-                    <mainClass>org.apache.flink.tools.ci.licensecheck.LicenseChecker</mainClass>
                     <includePluginDependencies>true</includePluginDependencies>
                     <includeProjectDependencies>false</includeProjectDependencies>
                 </configuration>
                 <dependencies>
                     <dependency>
                         <groupId>org.apache.flink</groupId>
                         <artifactId>flink-ci-tools</artifactId>
-                        <version>1.17-SNAPSHOT</version>
+                        <version>1.18-SNAPSHOT</version>
                     </dependency>
                 </dependencies>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```

Comment:
why is this needed?

Reply:
that's the requirement to make `ShadeOptionalChecker` working
previous versions of `maven-dependency-plugin` do not mark dependencies `optional` in output of `mvn dependency:tree`